events/eventconsumer/event_consumer.go

```diff
@@ -84,32 +72,22 @@ func (e EventConsumer) Validate() error {
 	return nil
 }
 
-// ConsumeEventsUntilEmpty consumes events forever until the event queue is empty.
+// ConsumeEventsUntilEmpty consumes events in a loop until the event queue is empty.
 func (t *EventConsumer) ConsumeEventsUntilEmpty(ctx *api.Context) {
-	consumerFunc := func(wg *sync.WaitGroup) {
-		for {
-			err := t.consumerFunc(ctx)
-			if err != nil {
-				if api.ErrorCode(err) == api.ENOTFOUND {
-					wg.Done()
-					return
-				}
-
-				logger.Errorf("error processing event, waiting %s to try again: %v", waitDurationOnFailure, err)
-				time.Sleep(waitDurationOnFailure)
+	for {
```

Comment:
Isn't this an endless loop if there is nothing to consume?

Reply:
If there's nothing to consuming we return `api.ENOTFOUND` which terminates this loop.

Reply:
I don't think we should be using an error for that - in some places we use it when a resource id is not found (an error) and here we are using it for an expected condition - Its probably better to return the number of events consumed